### ISSUE TITLE
Sync stalls due to repeatedly syncing snapshot. 

### ISSUE THREAD

**Issue by taoeffect** [openethereum/openethereum#10226]
_Before filing a new issue, please **provide the following information**._

- **Parity Ethereum version**: Parity-Ethereum/v2.2.7-stable-b00a21f-20190115/x86_64-linux-gnu/rustc1.31.1
- **Operating system**: Linux
- **Installation**: latest Docker stable
- **Fully synchronized**: ?? see comments below
- **Network**: ethereum
- **Restarted**: yes

_Your issue description goes here below. Try to include **actual** vs. **expected behavior** and **steps to reproduce** the issue._

```
6, 35, 38, 130, 212, 58, 85, 150, 173, 93, 4, 8, 216, 44, 187, 64, 120, 45, 219, 203, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 4, 188, 63, 166] }, v: 37, r: 106015295811830243713426629236258711318502398628365237097701616171120618638150, s: 55747325671259872567021190038153792588749884013945446278766325460757638285230, hash: 0x66ddd1a83417c1f2fc6d6ec39a3a0763ad729044d08ab5b7588d9ba04788f34b }
[Tx 51] UnverifiedTransaction { unsigned: Transaction { nonce: 1, gas_price: 20000000000, gas: 120000, action: Call(0x9f8646a35db0f466ac9322e2d194cc18f209fc75), value: 151176000000000000, data: [] }, v: 27, r: 105144428566320785931006985865370800312878258450319374579164336480691232519009, s: 37349750881345651839771539787728695041946700615631149944413910564829981544132, hash: 0x2562c4103262df4f70adf0db6abe6cc28f8326f05d5e5e7e0ea9be00415ace1e }
[Tx 52] UnverifiedTransaction { unsigned: Transaction { nonce: 42, gas_price: 20000000000, gas: 120000, action: Call(0x9f8646a35db0f466ac9322e2d194cc18f209fc75), value: 249485942821241676, data: [] }, v: 27, r: 37757209542033242717166078160944480511431263375434055827884008479592866928001, s: 52950977319609561404214233102531642145265306014619723050075751475648543934893, hash: 0x7a93913b9d722ee1792cbf29972f79fdedc0b3a0e319dcfb3baa022bccccd0c0 }
[Tx 53] UnverifiedTransaction { unsigned: Transaction { nonce: 26, gas_price: 20000000000, gas: 120000, action: Call(0x9f8646a35db0f466ac9322e2d194cc18f209fc75), value: 17556343960000000000, data: [] }, v: 27, r: 77712768997788904390940022323187624982915604954373242052406408588221844461069, s: 11044913212457021960395223175156154914557575898768655296491273777679066633739, hash: 0x394d40e16fbd5aa4e7f343167159c2e79b2d021843de3a21afd97a3ee21d316c }
[Tx 54] UnverifiedTransaction { unsigned: Transaction { nonce: 31425, gas_price: 20000000000, gas: 70000, action: Call(0x0e50e6d6bb434938d8fe670a2d7a14cd128eb50f), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 254, 84, 134, 195, 254, 83, 109, 231, 104, 230, 231, 252, 63, 167, 53, 209, 2, 127, 10, 136, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 122, 202, 11, 115] }, v: 37, r: 38383874623729549501937652666807916060803542365508177362292517256407889207487, s: 20881672966560400455876680165984125708721259525790692819764455579562202529172, hash: 0x4c30a21712a58eef168e2a65ce90d725e0d4d54bd3b290e64eae0d8aae1d5397 }
[Tx 55] UnverifiedTransaction { unsigned: Transaction { nonce: 31629, gas_price: 20000000000, gas: 70000, action: Call(0x0e50e6d6bb434938d8fe670a2d7a14cd128eb50f), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 211, 139, 146, 91, 99, 199, 157, 221, 29, 118, 81, 246, 156, 49, 1, 74, 39, 247, 35, 203, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 205, 142, 145, 43] }, v: 37, r: 32048200326757565061583990190379094102820305540263098798676191755595305096662, s: 35631095058531340720397145401773426590277747070415909710281985575566266948701, hash: 0xe24ffd3619a91f2875bd0bc0c2b0579a0d758f0577f144a531f52b680cd18496 }
[Tx 56] UnverifiedTransaction { unsigned: Transaction { nonce: 35063, gas_price: 20000000000, gas: 70000, action: Call(0x0e50e6d6bb434938d8fe670a2d7a14cd128eb50f), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 197, 137, 189, 58, 232, 44, 56, 41, 66, 96, 251, 107, 38, 42, 158, 3, 171, 34, 84, 55, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 69, 218, 81, 26, 58] }, v: 37, r: 81559235035442479393902921452095871000815463240297641425235243099929742693105, s: 35568955131155265616698072586790686341657426166454736912616969333337808808287, hash: 0x76fc84c71b4301cb394b4dcbcb2ac4c06d49f93290badd0594e9fd42d8c3b6cf }
[Tx 57] UnverifiedTransaction { unsigned: Transaction { nonce: 29361, gas_price: 20000000000, gas: 70000, action: Call(0x0e50e6d6bb434938d8fe670a2d7a14cd128eb50f), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 48, 195, 120, 21, 69, 11, 102, 9, 198, 27, 194, 176, 14, 88, 91, 162, 239, 236, 46, 214, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 124, 160, 230, 153] }, v: 38, r: 98322827142257868681919619825187040240003859667056380187049329372964913060424, s: 506376778568404090684341762058413404677576274059645113953432058484142752860, hash: 0x1209c8d408477fd9dd3a9b097445caea5d75a82a8ac8d49ef729888a424678d8 }
[Tx 58] UnverifiedTransaction { unsigned: Transaction { nonce: 97, gas_price: 20000000000, gas: 113150, action: Call(0xe4bd56cbf537074e3836a1721983107cce9e689f), value: 0, data: [141, 24, 47, 1, 220, 188, 176, 216, 92, 3, 219, 107, 133, 94, 66, 95, 203, 33, 159, 100, 53, 160, 65, 72, 121, 57, 186, 187, 86, 162, 90, 70, 175, 197, 96, 0] }, v: 37, r: 115553241185684619014189839772364573726004890709053838739206204351454943832772, s: 8205618680761607061142507651407166981572081438619461437369806329816233529802, hash: 0x04c50d87e33d4c60e15566569fc759bc10a7a5a352b2d89437d1b8749bc4ce79 }
[Tx 59] UnverifiedTransaction { unsigned: Transaction { nonce: 75, gas_price: 18000000000, gas: 210000, action: Call(0xc9c4d9ec2b44b241361707679d3db0876ac10ca6), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 118, 45, 162, 237, 81, 136, 241, 23, 233, 147, 127, 197, 47, 117, 113, 228, 11, 30, 146, 59, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 15, 12, 240, 100, 221, 89, 32, 0, 0] }, v: 28, r: 27467913381455113419162862914649335747780003672490789195671738180731904085226, s: 8580470914830034525814090766727131051164406318113399882133485158346137532405, hash: 0xbc1c526b3de641efa68e067d826e4908ecdbf164b766a494497ee7a8c513126f }
[Tx 60] UnverifiedTransaction { unsigned: Transaction { nonce: 87, gas_price: 18000000000, gas: 210000, action: Call(0xc9c4d9ec2b44b241361707679d3db0876ac10ca6), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 243, 12, 185, 230, 244, 140, 222, 95, 61, 241, 50, 49, 212, 102, 253, 133, 226, 104, 126, 124, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 54, 53, 201, 173, 197, 222, 160, 0, 0] }, v: 28, r: 34577142158423983888334382921632522129334793173053020170642347664435280215270, s: 42562913191078533808369555185289263354368487940598967118610288043922315344862, hash: 0x9efe6e8f300fdff42cb49af6d1569c3d23210a725dbd587f1720f86e9e474777 }
[Tx 61] UnverifiedTransaction { unsigned: Transaction { nonce: 38, gas_price: 18000000000, gas: 210000, action: Call(0x43030fe232ff28b628c58c1466e9238487f2d670), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 248, 155, 69, 43, 210, 47, 248, 65, 175, 245, 57, 25, 32, 16, 35, 13, 20, 55, 202, 101, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 42, 90, 5, 128, 214, 181, 58, 34, 173, 0] }, v: 37, r: 41323719521696238592321790263826661074974571387733984738016057796534566060414, s: 20658690042165011665346894468277006022577524669479779432330796164401326794598, hash: 0x2a03e9b27b967319c2c6400bcd2013d1f72cee8841f856b6696b00912a196b25 }
[Tx 62] UnverifiedTransaction { unsigned: Transaction { nonce: 11, gas_price: 18000000000, gas: 210000, action: Call(0xc9c4d9ec2b44b241361707679d3db0876ac10ca6), value: 0, data: [169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 162, 163, 95, 90, 100, 99, 245, 1, 205, 124, 50, 52, 134, 168, 52, 66, 117, 109, 60, 141, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 15, 12, 240, 100, 221, 89, 32, 0, 0] }, v: 28, r: 50098311076218656213100246195299469946650026371857822788321462641975923972092, s: 25822002589668025721691662770308985522807952603055242117812711259720599228083, hash: 0xfc2fbcb83c2ba0909136bd7ad3dbc0c4a54d2b4c4abf3847eb08aa9469974fee }

2019-01-21 07:44:49  Syncing #7104397 0xd32e…22a5     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed  #7104389    7/25 peers   6 MiB chain 125 MiB db 0 bytes queue 3 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

That's only one example. Right now there are over 150 of them.

And every time I restart parity it does this syncing snapshot business that takes forever:

```
2019-01-21 07:46:04  Syncing snapshot 2/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:09  Syncing snapshot 23/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:14  Syncing snapshot 45/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:19  Syncing snapshot 59/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:24  Syncing snapshot 73/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:29  Syncing snapshot 80/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:34  Syncing snapshot 87/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:39  Syncing snapshot 94/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:44  Syncing snapshot 102/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:49  Syncing snapshot 108/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-01-21 07:46:54  Syncing snapshot 115/2635        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

**Comment by joshua-mir** [comment]
Re: the unverified transactions -  you might be getting transactions from the network that stay in your txpool - they will remain unverified until you are in sync or they are deemed invalid. You can flush the current set of transactions by restarting with `--no-persistent-txqueue`, however this is only a temporary solution. You can also tweak your txpool/mining settings to accept fewer transactions.

Re: syncing snapshot: that means your node is performing a warp sync, which really shouldn't be happening on every single restart. Running with `--no-warp` should prevent it from happening again in the future.

**Comment by 5chdn** [comment]
I can reproduce this. It entirely kills the sync process and gets stuck

**Comment by taoeffect** [comment]
Those of you who are encountering this and are considering following @joshua-mir's advice, I just want you to know that it's been 4 days since I ran parity with those two flags and my node is still not finished syncing (on beefy hardware).

**Comment by TabiasP** [comment]
I've been stuck on snapshot #1089 for a few hours now.

**Comment by taoeffect** [comment]
After all that effort, I'm back to syncing the snapshot on each restart:

```
2019-02-10 11:22:11  Starting Parity-Ethereum/v2.2.7-stable-b00a21f-20190115/x86_64-linux-gnu/rustc1.31.1
2019-02-10 11:22:11  Keys path /home/parity/.local/share/io.parity.ethereum/keys/ethereum
2019-02-10 11:22:11  DB path /home/parity/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2019-02-10 11:22:11  State DB configuration: fast
2019-02-10 11:22:11  Operating mode: active
2019-02-10 11:22:11  Configured for Ethereum using Ethash engine
2019-02-10 11:22:11  Failed to auto-update latest ETH price: Fetch(Hyper(Error { kind: Connect, cause: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
2019-02-10 11:22:17  Public node URL: enode://8050b64c12abd39a6d4dda227435bb4b7bb3dd64add9d850250a03874ef15cafe3d0283ac1ec4f0afa913a9aef3399306937e2cf9461671cfaacb957813375a2@172.19.0.3:54322
2019-02-10 11:22:17  Syncing snapshot 3/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:21  Syncing snapshot 14/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:26  Syncing snapshot 37/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:31  Syncing snapshot 56/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:36  Syncing snapshot 68/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:41  Syncing snapshot 74/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:47  Syncing snapshot 82/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:51  Syncing snapshot 87/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:22:56  Syncing snapshot 94/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:01  Syncing snapshot 101/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:06  Syncing snapshot 108/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:11  Syncing snapshot 114/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:17  Syncing snapshot 121/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:22  Syncing snapshot 127/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:30  Syncing snapshot 132/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:23:34  Syncing snapshot 134/2694        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

That was preceded by the full node losing all of its connections (this is why I had to restart it):

```
2019-02-10 11:11:40     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:12:10     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:12:40     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:13:10     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:13:40  Syncing #7203613 0x2970…d5e6     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed  #7203534    1/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:14:10     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:14:40     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:15:10     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:15:40     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:16:10     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-02-10 11:16:40     0/25 peers   6 MiB chain 263 MiB db 0 bytes queue 10 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

Which in turn was preceded by a bunch of rubbish that is too long to paste and you can read here for up to 2 weeks before the pastie expires: https://pastebin.com/hr9m4E0q

**Comment by SjonHortensius** [comment]
Using **Parity-Ethereum/v2.4.6-stable-94164e1ca5-20190514/x86_64-linux-gnu/rustc1.34.1** this still happens. Strangely enough I do seem to connect to the maxium # peers, not the default 25

```
2019-06-12 10:15:16  Syncing snapshot 1489/3256  #4880000   306/2500 peers    126 KiB chain    1 KiB db  0 bytes queue  241 KiB sync  RPC:  0
2019-06-12 10:15:21  Syncing snapshot 1494/3256  #4880000   308/2500 peers    126 KiB chain    1 KiB db  0 bytes queue  241 KiB sync  RPC:  0
2019-06-12 10:15:26  Syncing snapshot 1498/3256  #4880000   308/2500 peers    126 KiB chain    1 KiB db  0 bytes queue  241 KiB sync  RPC:  0
2019-06-12 10:15:31  Syncing snapshot 1498/3256  #4880000   309/2500 peers    126 KiB chain    1 KiB db  0 bytes queue  241 KiB sync  RPC:  0
...
2019-06-27 11:46:53  Syncing snapshot 1498/3256  #4880001   2401/2500 peers    106 MiB chain  313 KiB db  0 bytes queue    2 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-06-27 11:46:58  Syncing snapshot 1498/3256  #4880001   2401/2500 peers    106 MiB chain  313 KiB db  0 bytes queue    2 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-06-27 11:47:03  Syncing snapshot 1498/3256  #4880001   2401/2500 peers    106 MiB chain  313 KiB db  0 bytes queue    2 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-06-27 11:47:08  Syncing snapshot 1498/3256  #4880001   2401/2500 peers    106 MiB chain  313 KiB db  0 bytes queue    2 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-06-27 11:47:13  Syncing snapshot 1498/3256  #4880001   2401/2500 peers    106 MiB chain  313 KiB db  0 bytes queue    2 MiB sync  RPC:  0 c
```

How am I supposed to recover, restart until it works?

**Comment by dvdplm** [comment]
@SjonHortensius If the syncing fails it will retry when you restart, so the "interesting" part comes at the very end. Can you enable logging with `-lsnapshot=trace` and let the sync finish? It is possible that something odd is happening at the very end. While importing the snapshot the number of peers is probably not relevant: your node is not yet fully operating.

**Comment by SjonHortensius** [comment]
@dvdplm After restarting with `-lsnapshot=trace`, unfortunately this works correctly now. I'll leave the trace enabled in case this happens again

**Comment by dvdplm** [comment]
> unfortunately this works correctly now.

yeah, these issues can be pesky to debug

> I'll leave the trace enabled in case this happens again

Thank you, much appreciated.

**Comment by taoeffect** [comment]
OK, so a few weeks ago I downloaded `v2.5.9-stable` in a docker container and ran it with `-lsnapshot=trace`.

It took several weeks to finish syncing. I saved the entire output to a file, which is 19GB.

Here's the first 200 lines from that file: https://pste.me/#/0BfTV

I can give more if needed. Please let me know soon though because I don't want to keep a 19GB text file around for very long if not needed.

I recently updated to the stable parity docker image. I'm not sure which version that corresponds to but it's the one that was built from [this Dockerfile at this commit](https://github.com/paritytech/parity-ethereum/blob/06c7096054764ccca1672182be4de98372cc7ebb/scripts/docker/hub/Dockerfile) on `"2019-09-30T16:27:22Z"` according to the `"io.parity.image.created"` and `"io.parity.image.source"` labels.

I ran it a few days ago and it's again doing what I don't know, but it doesn't seem to be picking up from where it left off. Here's what it's currently outputting to the screen after running for a couple of days: https://pste.me/#/55P2g

Basically, I can't tell exactly what this means, but I think it means my node is completely broken and in a serious way? Is that right? FYI, this node has not been used to send or receive any transactions by me or anyone else, only relay existing transactions on the network, so... it's basically a clean/fresh unmolested node inside of a docker container.

**Comment by taoeffect** [comment]
Also, for whatever reason the node is [usually connected to 0 peers](https://pste.me/#/55P2g). Sometimes it connects to one peer, but then it drops back down to 0.

This shouldn't be happening, right...? Anyway to figure out what's going on?

**Comment by taoeffect** [comment]
Oops, I had a totally separate problem with the second log output, which was that the computer clock was way off.

Here's the output of a recent run with `-lsnapshot=trace` on, running v2.5.9-stable: https://pastebin.com/6A2VxaXM

Two things of note:

1. The ` Syncing snapshot` issue is back
2. I am able to connect to 4 peers this time (assuming due to the fixed clock)

**Comment by taoeffect** [comment]
OK, now it doesn't seem to be syncing everything from the beginning, but it's still being weird. I can't tell from this if it's at 3039214 or 8900226...

```
2019-11-08 19:53:04    #3024227    1/25 peers   6 MiB chain 29 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:53:15  Imported #8900207 0x176b…a696 (154 txs, 7.71 Mgas, 654 ms, 25.63 KiB) + another 1 block(s) containing 0 tx(s)
2019-11-08 19:53:34  Reorg to #8900208 0x3fd0…f8d8 (0x176b…a696 #8900206 0x5b1d…87b5 0x511b…d51e)
2019-11-08 19:53:34  Imported #8900208 0x3fd0…f8d8 (174 txs, 8.59 Mgas, 506 ms, 28.71 KiB)
2019-11-08 19:53:34    #3025497    1/25 peers   5 MiB chain 29 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:53:39  Syncing #8900208 0x3fd0…f8d8     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed  #8900208    1/25 peers   5 MiB chain 29 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:54:09  Syncing #8900208 0x3fd0…f8d8     0.03 blk/s    6.4 tx/s    0.3 Mgas/s      0+    0 Qed  #8900211    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:54:09  Reorg to #8900209 0x57c2…c44c (0x3fd0…f8d8 #8900207 0x511b…d51e 0xfe52…3f99)
2019-11-08 19:54:10  Imported #8900211 0xf572…5fad (79 txs, 9.94 Mgas, 761 ms, 14.46 KiB) + another 2 block(s) containing 197 tx(s)
2019-11-08 19:54:39    #3028418    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:55:09    #3030196    1/25 peers   6 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:55:11  Imported #8900212 0xa028…101b (261 txs, 9.92 Mgas, 721 ms, 41.09 KiB)
2019-11-08 19:55:14  Imported #8900213 0xf280…254f (153 txs, 9.93 Mgas, 439 ms, 32.48 KiB)
2019-11-08 19:55:18  Imported #8900214 0x74f9…33e0 (213 txs, 9.94 Mgas, 481 ms, 35.83 KiB)
2019-11-08 19:55:36  Imported #8900215 0xac62…3d7f (24 txs, 1.05 Mgas, 80 ms, 4.67 KiB)
2019-11-08 19:55:39    #3031466    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:55:42  Imported #8900216 0x6bc2…e6a6 (0 txs, 0.00 Mgas, 3 ms, 0.53 KiB)
2019-11-08 19:56:04  Syncing #8900217 0xd4d7…c14e     0.08 blk/s    8.7 tx/s    0.4 Mgas/s      0+    0 Qed  #8900217    1/25 peers   7 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:56:10  Imported #8900218 0x6f51…f203 (200 txs, 9.93 Mgas, 720 ms, 32.96 KiB) + another 1 block(s) containing 218 tx(s)
2019-11-08 19:56:24  Imported #8900219 0x1917…1038 (125 txs, 5.23 Mgas, 359 ms, 20.97 KiB)
2019-11-08 19:56:30  Imported #8900220 0xf3e7…8032 (69 txs, 9.92 Mgas, 282 ms, 12.31 KiB)
2019-11-08 19:56:34    #3034514    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:56:55  Imported #8900221 0xb388…8dc2 (113 txs, 9.94 Mgas, 408 ms, 20.30 KiB)
2019-11-08 19:57:04    #3036420    1/25 peers   6 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:57:34    #3037689    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:57:39  Syncing #8900222 0x07ba…1bad     0.20 blk/s   51.2 tx/s    2.0 Mgas/s      0+    0 Qed  #8900222    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:58:03  Imported #8900223 0x0861…8bec (272 txs, 9.92 Mgas, 532 ms, 42.37 KiB) + another 1 block(s) containing 256 tx(s)
2019-11-08 19:58:08  Imported #8900225 0xfe3f…686f (70 txs, 9.93 Mgas, 353 ms, 13.19 KiB)
2019-11-08 19:58:09    #3039214    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:58:20  Imported #8900226 0x4b92…0298 (150 txs, 9.93 Mgas, 496 ms, 25.25 KiB)
2019-11-08 19:58:39    #3040737    1/25 peers   5 MiB chain 30 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:59:09    #3042135    1/25 peers   6 MiB chain 31 MiB db 0 bytes queue 15 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2019-11-08 19:59:25  Imported #8900228 0x0783…0d12 (182 txs, 6.97 Mgas, 461 ms, 29.91 KiB) + another 1 block(s) containing 166 tx(s)
2019-11-08 19:59:39    #3043658    1/25 peers   5 MiB chain 31 MiB db 0 bytes queue 14 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

It's also scary that it's connected to a single peer only.
